.github/actions/build/action.yml

```diff
@@ -0,0 +1,70 @@
+name: 'Build'
+description: 'Builds solc executable.'
+inputs:
+  release-suffix:
+    description: 'Release suffix.'
+    required: false
+    default: ''
+  release-version:
+    description: 'Release version.'
+    required: true
+  extra-args:
+    description: 'Extra arguments for cmake.'
+    required: false
+    default: ''
+runs:
+  using: "composite"
+  steps:
+
+    - name: Install boost
+      if: runner.os == 'macOS'
+      shell: zsh {0}
+      run: brew install boost
+
+    - name: Install Boost
+      if: runner.os == 'Windows'
+      shell: 'msys2 {0}'
+      env:
+        MSYS_BOOST_PACKAGE: 'mingw-w64-x86_64-boost-1.79.0-4-any.pkg.tar.zst'
+      run: |
+        pacman -Sy
+        curl -LO "https://repo.msys2.org/mingw/mingw64/${MSYS_BOOST_PACKAGE}"
+        pacman --noconfirm -U "${MSYS_BOOST_PACKAGE}"
+
+    - name: Build solc
+      shell: ${{ runner.os == 'Windows' && 'msys2 {0}' || 'bash' }}
+      run: |
+        mkdir -p ./build
+        cd ./build
+        cmake .. \
+          -DCMAKE_BUILD_TYPE="Release" \
+          -DSOLC_VERSION_ZKSYNC="${{ inputs.release-version }}" \
+          -DUSE_Z3=OFF \
+          -DUSE_CVC4=OFF ${{ inputs.extra-args}}
+        cmake --build . --config Release -j 10
```

Comment:
I see the "10" in the original as well. Is it possible to not limit the job count here? (I'm not sure if just "-j" in cmake --build is same as just "-j" in make).

Reply:
True, let's not hardcode it and look up the thread count.

Reply:
This one is tricky... even `make -j` is not working by default on all platforms and requires the number of threads to be explicitly specified. At the same time, defining the number of CPU threads for all platforms differs. I'll think about something universal or introduce a new platform-specific step.

Reply:
Let's do properly with a platform specific step. On Linux and MacOS it's easy, and for Windows... hardcode it deserves :)

Reply:
Completed.